drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,229 @@
+by block height checks. For example:
+
+.. code:: cpp
+
+  if (CurrentEpoch(chainActive.Height(), Params().GetConsensus()) == Consensus::UPGRADE_OVERWINTER) {
+      // Overwinter-specific logic
+  } else {
+      // Non-Overwinter logic
+  }
+
+  // ...
+
+  if (NetworkUpgradeActive(pindex->nHeight, Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
+      // Overwinter consensus rules applied to block
+  } else {
+      // Pre-Overwinter consensus rules applied to block
+  }
+
+
+Block validation
+````````````````
+Incoming blocks known to have a particular height (due to their parent chain being entirely known) MUST be
+validated under the consensus rules corresponding to the expected branch ID for that height.
+
+Incoming blocks with unknown heights (because at least one block header in their parent chain is unknown)
+MAY be cached for future consideration after all their parents have been received.
+
+Chain reorganization
+````````````````````
+It is possible for a reorganization to occur that rolls back from after the activation height, to before that
+height. This can handled in the same way as any regular chain orphaning or reorganization, as long as the new
+chain is valid.
+
+Post-activation upgrading
+`````````````````````````
+If a user does not upgrade their node to a compatible software version before ``ACTIVATION_HEIGHT`` is
+reached, their node will follow any pre-upgrade branch that persists, and may download blocks that are
+incompatible with the post-upgrade branch. If the user subsequently upgrades their node to a compatible
+software version, the node will consider these blocks to be invalid, and MUST take one of the two following
+actions:
+
+- Discard all blocks of height ``ACTIVATION_HEIGHT`` and above, and then synchronize with the network.
+
+- Shut down and alert the user of the issue. In this case, the node could offer an option to perform the first
+  action.
+
+Memory pool
+-----------
+
+While the current chain tip height is below ``ACTIVATION_HEIGHT``, nodes SHOULD NOT accept transactions that
+will only be valid on the post-upgrade branch.
+
+When the current chain tip height reaches ``ACTIVATION_HEIGHT``, the node's local transaction memory pool
+SHOULD be cleared of transactions that will never be valid on the post-upgrade branch.
+
+Two-way replay protection
+-------------------------
+
+Before the Overwinter network upgrade, two-way replay protection is ensured by enforcing post-upgrade that the
+most significant bit of the transaction version is set to 1. [#zip-tx-format]_ From the perspective of old
+nodes, the transactions will have a negative version number, which is invalid under the old consensus rules.
+Enforcing this rule trivially makes old transactions invalid on the Overwinter branch.
+
+After the Overwinter network upgrade, two-way replay protection is ensured by transaction signatures
+committing to a specific ``BRANCH_ID``. [#zip-0143]_
```

Comment:
It is not clear from the description if the ``branch_id`` committed to here needs to be the same as the branch_id appearing in the transaction from version 3 as described in transaction format zip  https://github.com/bitcartel/zips/blob/d447e0db0177f461b0020aa2560d6b4ca4234578/drafts/bitcartel-nu0-tx-format/draft.rst
I think this is not enforced. A potential problem with this committing in the signature is that what about transactions with no signature, like a coin base one.

Reply:
This related to my comment here https://github.com/zcash/zcash/pull/2903/files#r167205498

Reply:
I think the renaming of the field in the transaction to ``nVersionGroupId`` sufficiently addresses this comment.